### ISSUE TITLE
Unlatex broken for https://beta.lmfdb.org/NumberField/4.0.1008.2

### ISSUE THREAD

**Issue by edgarcosta** [LMFDB/lmfdb#5070]
```
Exception on /NumberField/4.0.1008.2/download/data [GET]
Traceback (most recent call last):
  File "/home/sage/sage-9.4/local/lib/python3.9/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/sage/sage-9.4/local/lib/python3.9/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/sage/sage-9.4/local/lib/python3.9/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/sage/sage-9.4/local/lib/python3.9/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/sage/sage-9.4/local/lib/python3.9/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/sage/sage-9.4/local/lib/python3.9/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/lmfdb/lmfdb-git-dev/lmfdb/number_fields/number_field.py", line 988, in nf_download
    response = make_response(nf_data(**args))
  File "/home/lmfdb/lmfdb-git-dev/lmfdb/number_fields/number_field.py", line 1006, in nf_data
    units = str(unlatex(nf.units()))
  File "/home/lmfdb/lmfdb-git-dev/lmfdb/number_fields/number_field.py", line 973, in unlatex
    s = re.sub(r'\\+', r'\\',s)
  File "/home/sage/sage-9.4/local/lib/python3.9/re.py", line 210, in sub
    return _compile(pattern, flags).sub(repl, string, count)
TypeError: expected string or bytes-like object
[2022-03-11 13:06:38 UTC] 500 error on URL https://beta.lmfdb.org/NumberField/4.0.1008.2/download/data ()
```

**Comment by jwj61** [comment]
It looks like the actual error is in nf_data.  It seems to assume units are in the database (and nf.units() might be None), it needs to apply unlatex to each entry of the list, and then not add extra brackets at the end.

**Comment by AndrewVSutherland** [comment]
Resolved via #5079
